src/page-analysis/background/get-fav-icon.js

```diff
@@ -1,20 +1,26 @@
+import favicon from 'favicon'
+
 import responseToDataURI from 'src/util/response-to-data-uri'
 
-// Get a tab's fav-icon (website logo) as a data URI
-async function getFavIcon({tabId}) {
-    const tab = await browser.tabs.get(tabId)
+// Async wrapper around favicon package
+const fetchFaviconUrl = url =>
+    new Promise((resolve, reject) =>
+        favicon(url, (err, favIconUrl) => err ? reject(err) : resolve(favIconUrl)))
```

Comment:
You might like to use [pify](https://www.npmjs.com/package/pify) (promisify) to just say `pify(favicon)` instead.